src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -532,33 +524,37 @@ GUID CascadiaSettings::_GetProfileForArgs(const NewTerminalArgs& newTerminalArgs
         {
             bool wasGuid = false;
 
-            try
+            // Do a quick heuristic check - is the profile 38 chars long (the
+            // length of a GUID string), and does it start with '{'? Because if
+            // it doesn't, it's _definitely_ not a GUID.
+            if (newTerminalArgs.Profile().size() == 38 && newTerminalArgs.Profile()[0] == L'{')
             {
-                const auto newGUID = Utils::GuidFromString(newTerminalArgs.Profile().c_str());
-
-                for (const auto& p : _profiles)
+                try
                 {
-                    if (p.GetGuid() == newGUID)
+                    const auto newGUID = Utils::GuidFromString(newTerminalArgs.Profile().c_str());
+
+                    for (const auto& p : _profiles)
                     {
-                        profileGuid = newGUID;
-                        wasGuid = true;
-                        break;
+                        if (p.GetGuid() == newGUID)
+                        {
+                            profileGuid = newGUID;
+                            wasGuid = true;
+                            break;
+                        }

```

Comment:
I feel like there's _also_ a function that finds a profile based on its guid that does this. Should we consolidate on that? It will leave us free to change the backing store implementation later.